lib/resources/platform.rb

```diff
@@ -0,0 +1,92 @@
+# encoding: utf-8
+
+module Inspec::Resources
+  class PlatformResource < Inspec.resource(1)
+    name 'platform'
+    desc 'Use the platform InSpec resource to test the platform on which the system is running.'
+    example "
+      describe platform do
+        its('name') { should eq 'redhat' }
+      end
+
+      describe platform do
+        it { should be_in_family('unix') }
+      end
+    "
+
+    def initialize
+      @platform = inspec.backend.os
+    end
+
+    # add helper methods for easy access of properties
+    %w{name family release arch}.each do |property|
+      define_method(property.to_sym) do
+        @platform.send(property)
+      end
+    end
+
+    def [](name)
+      # convert string to symbol
+      name = name.to_sym if name.is_a? String
+      @platform[name]
+    end
+
+    def platform?(name)
+      @platform.name == name ||
+        @platform.family_hierarchy.include?(name)
+    end
+
+    def in_family?(family)
+      @platform.family_hierarchy.include?(family)
+    end
+
+    def families
+      @platform.family_hierarchy
+    end
+
+    def supported?(supports)
```

Comment:
I really like that we have one method that rules them all. I am just not sure if this is not better suited for train than this resource.

Reply:
Yeah it could live in either. The only issue with train is we would be sending InSpec specific keys to the support library.

Reply:
I think InSpec is the right place for this logic to live. I see Train as the thing that surfaces the information about the platform, and provides the means to connect to it. Those that consume that information can decide whether what they intend to do is supported by the information Train has surfaced.

If @chris-rock feels strongly about moving this logic into Train, then this method could be transformed to something that puts together a cleaned-up hash of the Train-specific keys, and then passes them to a `supported?` method on the Train side.

However, I still think this lives within InSpec.

Reply:
Looks good here 👍 Agreed it could live in either and I like Adam's explanation too.